eating_cookies/eating_cookies.py

```diff
@@ -3,14 +3,77 @@
 import sys
 
 # The cache parameter is here for if you want to implement
-# a solution that is more efficient than the naive 
+# a solution that is more efficient than the naive
 # recursive solution
-def eating_cookies(n, cache=None):
-  pass
+
+
+def eating_cookies(n, cache={0: 1, 1: 1, 2: 2}):
+    # Base cases
+    # if there are invalid negative cookies, 0 ways
+    if n < 0:
+        return 0
+
+    if n not in cache:
+        cache[n] = eating_cookies(
+            n-1) + eating_cookies(n-2) + eating_cookies(n-3)
+
+    return cache[n]
+
+    # if n < 0:
+    #     return 0
+
+    # elif n == 1 or n == 0:
+    #     return 1
+
+    # elif n == 2:
+    #     return 2
+
+    # else:
+    #     return eating_cookies(n-1) + eating_cookies(n-2) + eating_cookies(n-3)
```

Comment:
Nice work finishing this. Looks like you implemented it without the cache first then added the cache. Always good to get to the answer first and not prematurely optimize, which you did!